### DISCUSSION TITLE
react query not using cached data on nextjs (small reproduce)

### DISCUSSION THREAD

**Discussion by isggwp** [TanStack/query#2386] [Q&A] [answered]
Hi mate,

i have using nextjs with react-query as SSR Mode and client mode. but fetcher always trigger every single moment in pages. its something wrong on my setup?

small reproduce:
https://github.com/qipoy/react-q-example

[![query-ex-1.gif](https://i.postimg.cc/TY2ZGQcJ/query-ex-1.gif)](https://postimg.cc/4nM8Q6qY)

**Comment by TkDodo** [comment, marked as answer]
It is using the cache, otherwise, you’d see a loading state between each navigation. It is also keeping your data up to date by doing background refetches. Have you read https://react-query.tanstack.com/guides/important-defaults ?

**Reply by isggwp** [reply]
even we have ```  isloading : true  ``` .  Does it still use cacahe?

**Reply by TkDodo** [reply]
if you get `isLoading: true` in between then no, it is not using the cache.

**Reply by isggwp** [reply]
ok I got it. thanks for the advice.
